src/_static/js/minicylc.js

```diff
@@ -0,0 +1,294 @@
+/*eslint no-console: off*/
+
+// Default Cylc colour theme.
+var minicylc_default_theme = {
+    'waiting_fill': 'none',
+    'waiting_stroke': '#88c6ff',
+    'running_fill': '#00c410',
+    'running_stroke': 'black',
+    'succeed_fill': '#ada5a5',
+    'succeed_stroke': 'black'
+}
+
+// Demo colour theme for demmonstrating workflow logic.
+var minicylc_demo_theme = {
+    'succeed_fill': '#aabbff',
+    'succeed_stroke': 'black'
+}
+
+
+class MiniCylc {
+    /**
+     * Class for animating SVG graphs.
+     *
+     * Attributes:
+     *   - nodes: A dictionary of task names against a list of SVG nodes.
+     *   - edges: A dictionary of task edges against a list SVG nodes.
+     *   - dependencies: A dictionary of task names against a list of
+     *     conditional expressions.
+     */
+
+    constructor(div) {
+        /**
+         * Initiate the object.
+         * @param div The <div> element containing the SVG.
+         * */
+
+        // Obtain nodes and edges from svg.
+        var svg = ($(div).find('object:first')[0]).contentDocument;
+        this._find_svg_elements(svg);
+
+        // Parse dependencies.
+        var deps = this._get_dependencies_from_graph(div);
+        this._construct_dependency_map(deps);
+
+        // Process colour theme.
+        this.setup_colours($(div).data('theme'));
+    }
+
+    setup_colours(theme) {
+        /**
+         * Set the colour theme.
+         * @param theme The name of a colour theme as a string.
+         */
+        if (!theme || theme == 'default') {
+            this.theme = minicylc_default_theme;
+        } else if (theme == 'demo') {
+            this.theme = minicylc_demo_theme;
+        } else {
+            console.log('Warning: Invalid theme detected "' + theme +
+                        '", defaulting to black.');
+            this.theme = {};
+        }
+    }
+
+    _find_svg_elements(svg) {
+        /**
+         * Associate task/dependency names with SVG nodes.
+         *
+         * Associations stored as dicionaries this.nodes and this.edges.
```

Comment:
s/dicionaries/dictionaries (got confused for a sec, that's almost the Portuguese spelling dicionarios)